client/src/components/Forms/AddNewSpending.js

```diff
@@ -60,7 +25,7 @@ const styleFormItem = {
   flex: '0 0 48%'
 };
 
-export default function SpendingForm({ handleClick, handleClose, value }) {
+export default function AddNewSpending({ handleClose, value }) {
```

Comment:
You will never call a property `handle...`. Instead, you call it `onClose`.

Only the actual function is called `handleClose`. 
See https://jaketrent.com/post/naming-event-handlers-react/